windowsforms/matching-game/net5-windows/vb/MatchingGame.Logic/Game.vb

```diff
@@ -0,0 +1,121 @@
+﻿Imports System.Collections.Generic
+
+Public NotInheritable Class Game
+
+    Private ReadOnly _cards() As Char
+    Private ReadOnly _openCards() As Boolean
+    Private ReadOnly _turnCards As New List(Of Integer)(2)
+
+    Private _turns As Integer
+
+    Private Sub New(width As Integer, height As Integer, board As Char())
+        Me.Width = width
+        Me.Height = height
+        _cards = board
+        ReDim _openCards(_cards.Length - 1)
+    End Sub
+
+    Public Shared Function Create() As Game
+        Dim numberOfPairs = 8
+        Dim width = numberOfPairs / 2
+        Dim height = width
+
+        Dim board((width * height) - 1) As Char
+        Dim availableCards = New List(Of Char) From
+        {
+            "!"c, "!"c, "N"c, "N"c, ","c, ","c, "k"c, "k"c,
+            "b"c, "b"c, "v"c, "v"c, "w"c, "w"c, "z"c, "z"c
+        }
+
+        Dim Random As New Random()
+
+        For i = 0 To numberOfPairs * 2 - 1
+            Dim cardIndex = random.Next(availableCards.Count - 1)
+            Dim card = availableCards(cardIndex)
+            board(i) = card
+            availableCards.RemoveAt(cardIndex)
+        Next
+
+        Return New Game(width, height, board)
+    End Function
+
+    Public ReadOnly Property Turns As Integer
+        Get
+            Return _turns
+        End Get
+    End Property
```

Comment:
Why this doesn't use auto-implemented property syntax like the `Width` and `Height` properties?

Reply:
I'm not sure. I didn't do a ton of cleanup on the original code. Considering this is supporting the migration article I don't think we'll modify this code unless there is a bug.